Fixes unsused flags from `tsh kube exec` command

tigrato opened the pull request:
Since its introduction, `tsh kube exec` didn't have any use for the `--reason` and `--invite` flags and completely ignored what users did with it.

This PR transports those flags as query parameters to Teleport Kubernetes Service which includes them when creating the session resource in Teleport.

codingllama commented on e:
Revert? I think this is downgrading e/.

tigrato replied:
`git add .` is a pain :upside_down_face:

codingllama replied:
Yep.

codingllama commented on lib/kube/proxy/sess.go:
Test server-side changes?

codingllama replied:
Thanks!

tigrato committed:
Fixes unsused flags from `tsh kube exec` command

Since its introduction, `tsh kube exec` didn't have any use for the `--reason` and `--invite` flags and completely ignored what users did with it.

This PR transports those flags as query parameters to Teleport Kubernetes Service which includes them when creating the session resource in Teleport.

tigrato committed:
add tests

codingllama commented on lib/kube/proxy/moderated_sessions_test.go:
```diff
@@ -429,3 +450,20 @@ func TestModeratedSessions(t *testing.T) {
+
+// validateSessionTracker validates that the session tracker has the expected
+// reason and invited users.
+func validateSessionTracker(t *testing.T, testCtx *TestContext, sessionID string, reason string, invited []string) error {
```
Drop the `t` param? I don't think we are using it. Or, alternative, drop the error response and use it?

tigrato committed:
fix flaky test
